security/fma-supervisor.md

```diff
@@ -0,0 +1,249 @@
+# OP-Supervisor: Failure Modes and Recovery Path Analysis
+
+| | |
+|--------|--------------|
+| Author | Axel Kingsley |
+| Created at | 2025-03-26 |
+| Needs Approval From | |
+| Other Reviewers |   |
+| Status | Draft |
+
+## Introduction
+
+This document covers the supervisor as a new consensus critical component, and by its nature involves interop protocol behaviors, as well as new behavior modalities in op-node.
+
+## Interop Protocol Terms and Behavior Explainer
+
+To start setting context, here is a basic overview of how Interop works at the protocol level:
+
+- **There are Two Chains, A and B** who participate in interop with one another.
+- Any time a log is emitted on A, it can be referenced on B.
+- And likewise, any logs on B can be referenced on A.
+- The way a chain can reference logs from other chains is by creating an *Executing Message*
+    - Executing Messages are emitted as Logs by the Cross-L2-Inbox contract when called.
+    - Each Executing Message contains indexing information and a hash.
+- By the rules of the protocol, Executing Messages are always valid on the canonical chain because:
+    - If they are valid, then they are valid.
```

Comment:
Could you elaborate what you mean here?

Reply:
I'll do a pass on the text in the FMA too, but what I mean by this is:

An `Executing Message` is a Log Event from this particular Smart Contract. *By the Rules Of the Protocol*, we know **all** `Executing Messages` fit some invariants (ie, they are valid). It is impossible for there to be an EM which *does not* fit the invariant, because if such a thing existed in a block, that entire block itself would be invalid.

Therefore, Executing Messages are either Valid, or if they are not Valid, the Block they exist in is non-canoncial (by way of replacement) anyway, so Invalid Messages don't ever really "exist" in the canonical chain.

"If they are Valid, then they are Valid" + "If they are Invalid, then they don't exist" = "Only Valid EMs Exist"